### ISSUE TITLE
[ERROR] RuntimeError: CUDA error: the launch timed out and was terminated (Driving me nuts)

### ISSUE THREAD

**Issue by PurpleBlueAloeVera** [comfyanonymous/ComfyUI#3380]
Ok, so here's the new bug that I'm getting and that I can't for the life of me fix. I reinstalled cuda, reinstalled ComfyUI.. updated my drivers (which are always up-to-date).. I just do NOT understand this error. It's driving me crazy.. can somebody help please.. 

``` 40%|█████████████████████████████████▏                                                 | 4/10 [00:41<01:02, 10.44s/it]
!!! Exception during processing!!! CUDA error: the launch timed out and was terminated
CUDA kernel errors might be asynchronously reported at some other API call, so the stacktrace below might be incorrect.
For debugging consider passing CUDA_LAUNCH_BLOCKING=1.
Compile with `TORCH_USE_CUDA_DSA` to enable device-side assertions.

Traceback (most recent call last):
  File "F:\AI_repos\ComfyUI\execution.py", line 151, in recursive_execute
    output_data, output_ui = get_output_data(obj, input_data_all)
  File "F:\AI_repos\ComfyUI\execution.py", line 81, in get_output_data
    return_values = map_node_over_list(obj, input_data_all, obj.FUNCTION, allow_interrupt=True)
  File "F:\AI_repos\ComfyUI\custom_nodes\ComfyUI_ezXY\autoCastPatch.py", line 299, in map_node_over_list
    return _map_node_over_list(obj, input_data_all, func, allow_interrupt)
  File "F:\AI_repos\ComfyUI\execution.py", line 74, in map_node_over_list
    results.append(getattr(obj, func)(**slice_dict(input_data_all, i)))
  File "F:\AI_repos\ComfyUI\nodes.py", line 1344, in sample
    return common_ksampler(model, seed, steps, cfg, sampler_name, scheduler, positive, negative, latent_image, denoise=denoise)
  File "F:\AI_repos\ComfyUI\nodes.py", line 1314, in common_ksampler
    samples = comfy.sample.sample(model, noise, steps, cfg, sampler_name, scheduler, positive, negative, latent_image,
  File "F:\AI_repos\ComfyUI\custom_nodes\ComfyUI-Impact-Pack\modules\impact\sample_error_enhancer.py", line 22, in informative_sample
    raise e
  File "F:\AI_repos\ComfyUI\custom_nodes\ComfyUI-Impact-Pack\modules\impact\sample_error_enhancer.py", line 9, in informative_sample
    return original_sample(*args, **kwargs)  # This code helps interpret error messages that occur within exceptions but does not have any impact on other operations.
  File "F:\AI_repos\ComfyUI\custom_nodes\ComfyUI-AnimateDiff-Evolved\animatediff\sampling.py", line 279, in motion_sample
    return orig_comfy_sample(model, noise, *args, **kwargs)
  File "F:\AI_repos\ComfyUI\custom_nodes\ComfyUI-Advanced-ControlNet\adv_control\control_reference.py", line 47, in refcn_sample
    return orig_comfy_sample(model, *args, **kwargs)
  File "F:\AI_repos\ComfyUI\comfy\sample.py", line 37, in sample
    samples = sampler.sample(noise, positive, negative, cfg=cfg, latent_image=latent_image, start_step=start_step, last_step=last_step, force_full_denoise=force_full_denoise, denoise_mask=noise_mask, sigmas=sigmas, callback=callback, disable_pbar=disable_pbar, seed=seed)
  File "F:\AI_repos\ComfyUI\comfy\samplers.py", line 755, in sample
    return sample(self.model, noise, positive, negative, cfg, self.device, sampler, sigmas, self.model_options, latent_image=latent_image, denoise_mask=denoise_mask, callback=callback, disable_pbar=disable_pbar, seed=seed)
  File "F:\AI_repos\ComfyUI\comfy\samplers.py", line 657, in sample
    return cfg_guider.sample(noise, latent_image, sampler, sigmas, denoise_mask, callback, disable_pbar, seed)
  File "F:\AI_repos\ComfyUI\comfy\samplers.py", line 644, in sample
    output = self.inner_sample(noise, latent_image, device, sampler, sigmas, denoise_mask, callback, disable_pbar, seed)
  File "F:\AI_repos\ComfyUI\comfy\samplers.py", line 623, in inner_sample
    samples = sampler.sample(self, sigmas, extra_args, callback, noise, latent_image, denoise_mask, disable_pbar)
  File "F:\AI_repos\ComfyUI\comfy\samplers.py", line 534, in sample
    samples = self.sampler_function(model_k, noise, sigmas, extra_args=extra_args, callback=k_callback, disable=disable_pbar, **self.extra_options)
  File "F:\AI_repos\stable-diffusion-webui-forge\venv\lib\site-packages\torch\utils\_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "F:\AI_repos\ComfyUI\comfy\k_diffusion\sampling.py", line 707, in sample_dpmpp_sde_gpu
    return sample_dpmpp_sde(model, x, sigmas, extra_args=extra_args, callback=callback, disable=disable, eta=eta, s_noise=s_noise, noise_sampler=noise_sampler, r=r)
  File "F:\AI_repos\stable-diffusion-webui-forge\venv\lib\site-packages\torch\utils\_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "F:\AI_repos\ComfyUI\comfy\k_diffusion\sampling.py", line 559, in sample_dpmpp_sde
    denoised_2 = model(x_2, sigma_fn(s) * s_in, **extra_args)
  File "F:\AI_repos\ComfyUI\comfy\samplers.py", line 272, in __call__
    out = self.inner_model(x, sigma, model_options=model_options, seed=seed)
  File "F:\AI_repos\ComfyUI\comfy\samplers.py", line 610, in __call__
    return self.predict_noise(*args, **kwargs)
  File "F:\AI_repos\ComfyUI\comfy\samplers.py", line 613, in predict_noise
    return sampling_function(self.inner_model, x, timestep, self.conds.get("negative", None), self.conds.get("positive", None), self.cfg, model_options=model_options, seed=seed)
  File "F:\AI_repos\ComfyUI\comfy\samplers.py", line 258, in sampling_function
    out = calc_cond_batch(model, conds, x, timestep, model_options)
  File "F:\AI_repos\ComfyUI\custom_nodes\ComfyUI-TiledDiffusion\.patches.py", line 4, in calc_cond_batch
    return calc_cond_batch_original_tiled_diffusion_d5cd7809(model, conds, x_in, timestep, model_options)
  File "F:\AI_repos\ComfyUI\comfy\samplers.py", line 218, in calc_cond_batch
    output = model.apply_model(input_x, timestep_, **c).chunk(batch_chunks)
  File "F:\AI_repos\ComfyUI\comfy\model_base.py", line 97, in apply_model
    model_output = self.diffusion_model(xc, t, context=context, control=control, transformer_options=transformer_options, **extra_conds).float()
  File "F:\AI_repos\stable-diffusion-webui-forge\venv\lib\site-packages\torch\nn\modules\module.py", line 1518, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "F:\AI_repos\stable-diffusion-webui-forge\venv\lib\site-packages\torch\nn\modules\module.py", line 1527, in _call_impl
    return forward_call(*args, **kwargs)
  File "F:\AI_repos\ComfyUI\comfy\ldm\modules\diffusionmodules\openaimodel.py", line 885, in forward
    h = forward_timestep_embed(module, h, emb, context, transformer_options, output_shape, time_context=time_context, num_video_frames=num_video_frames, image_only_indicator=image_only_indicator)
  File "F:\AI_repos\ComfyUI\comfy\ldm\modules\diffusionmodules\openaimodel.py", line 44, in forward_timestep_embed
    x = layer(x, context, transformer_options)
  File "F:\AI_repos\stable-diffusion-webui-forge\venv\lib\site-packages\torch\nn\modules\module.py", line 1518, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "F:\AI_repos\stable-diffusion-webui-forge\venv\lib\site-packages\torch\nn\modules\module.py", line 1527, in _call_impl
    return forward_call(*args, **kwargs)
  File "F:\AI_repos\ComfyUI\comfy\ldm\modules\attention.py", line 633, in forward
    x = block(x, context=context[i], transformer_options=transformer_options)
  File "F:\AI_repos\stable-diffusion-webui-forge\venv\lib\site-packages\torch\nn\modules\module.py", line 1518, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "F:\AI_repos\stable-diffusion-webui-forge\venv\lib\site-packages\torch\nn\modules\module.py", line 1527, in _call_impl
    return forward_call(*args, **kwargs)
  File "F:\AI_repos\ComfyUI\comfy\ldm\modules\attention.py", line 460, in forward
    return checkpoint(self._forward, (x, context, transformer_options), self.parameters(), self.checkpoint)
  File "F:\AI_repos\ComfyUI\comfy\ldm\modules\diffusionmodules\util.py", line 191, in checkpoint
    return func(*inputs)
  File "F:\AI_repos\ComfyUI\comfy\ldm\modules\attention.py", line 557, in _forward
    n = attn2_replace_patch[block_attn2](n, context_attn2, value_attn2, extra_options)
  File "F:\AI_repos\ComfyUI\custom_nodes\ComfyUI_IPAdapter_plus\CrossAttentionPatch.py", line 43, in __call__
    sigma = extra_options["sigmas"].detach().cpu()[0].item() if 'sigmas' in extra_options else 999999999.9
RuntimeError: CUDA error: the launch timed out and was terminated
CUDA kernel errors might be asynchronously reported at some other API call, so the stacktrace below might be incorrect.
For debugging consider passing CUDA_LAUNCH_BLOCKING=1.
Compile with `TORCH_USE_CUDA_DSA` to enable device-side assertions.


Prompt executed in 51.20 seconds
Exception in thread Thread-22 (prompt_worker):
Traceback (most recent call last):
  File "D:\python\lib\threading.py", line 1016, in _bootstrap_inner
    self.run()
  File "D:\python\lib\threading.py", line 953, in run
    self._target(*self._args, **self._kwargs)
  File "F:\AI_repos\ComfyUI\main.py", line 144, in prompt_worker
    comfy.model_management.soft_empty_cache()
  File "F:\AI_repos\ComfyUI\comfy\model_management.py", line 840, in soft_empty_cache
    torch.cuda.empty_cache()
  File "F:\AI_repos\stable-diffusion-webui-forge\venv\lib\site-packages\torch\cuda\memory.py", line 159, in empty_cache
    torch._C._cuda_emptyCache()
RuntimeError: CUDA error: the launch timed out and was terminated
CUDA kernel errors might be asynchronously reported at some other API call, so the stacktrace below might be incorrect.
For debugging consider passing CUDA_LAUNCH_BLOCKING=1.
Compile with `TORCH_USE_CUDA_DSA` to enable device-side assertions. ```

If anyone wants to comment on my venv. Yes, I intially was using a fresh ComfyUI venv, but still same issue. Here I tried to link an existing venv. But didn't help either..

**Comment by Declan-Bracken** [comment]
I'm having the same issue when trying to run inference on a quantized Mistral model. Drivers and pytorch version all up to date.

**Comment by EmersttaCreations** [comment]
I have the same issue too. It drives me nuts every single time I got the same error. I enable --lowvram to ComfyUI, but I still get the same error. I set the image latent to 768*768, but still get the same thing. The console always tells me to set `CUDA_LAUNCH_BLOCKING=1` before initiating an AI art.

But where can I set `CUDA_LAUNCH_BLOCKING=1` somewhere in ComfyUI?

**Comment by EngmidoAI** [comment]
NOBODY HAVE AN ANSWER

**Comment by RasheedOlaleye** [comment]
I am having the same issues

**Comment by Molinist** [comment]
welcome to the pond of scum... where we must all suffer and rot

**Comment by margo2130** [comment]
same problem, any solution?

**Comment by gatttahaveit1** [comment]
dump your error into Chat gpt. I'm sure your systems are different  than mine but Ai is really good at fixing problems.

Your GTX TITAN X (Maxwell, 12 GB) + PyTorch 2.7.0 (cu128) + WAN 2.2 i2v @ 1280×704 × 121 frames + UniPC is a perfect recipe for kernels that run longer than Windows’ default “watchdog” timeout. Windows then kills the CUDA launch with:

CUDA error: the launch timed out and was terminated

Below is a quick, battle-tested path to green.

1) First make the workload sane (quick win)

In your attached graph:

Wan22ImageToVideoLatent

width: 672

height: 384

length: 24 (start with 16–24; scale up after it works)

batch_size: 1

KSampler

steps: 8–12 (start 8)

cfg: 3–5 (start 3.5)

sampler_name: euler (or dpmpp_2m) — avoid uni_pc on older cards

scheduler: karras

denoise: 1

CreateVideo

fps: 12–16

These changes dramatically shorten kernels and reduce VRAM pressure.

2) Nudge Windows: raise TDR limits (prevents “launch timed out”)

Windows will still nuke long kernels unless you increase the timeout.

Create a file named increase_tdr.reg with this content, then double-click to import, reboot after:

Windows Registry Editor Version 5.00

[HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\GraphicsDrivers]
"TdrDelay"=dword:0000003c           ; 60 seconds
"TdrDdiDelay"=dword:0000003c        ; 60 seconds
; Optional (safer to leave enabled): "TdrLevel"=dword:00000003


Tip: I don’t recommend fully disabling TDR (TdrLevel=0). Raising the delay is usually enough.

3) Use a PyTorch/CUDA combo that likes Maxwell

Maxwell support and modern CUDA (12.x) can be rough. Your log shows 2.7.0+cu128; on Maxwell I’ve had far fewer hiccups with CUDA 11.8 builds (and ComfyUI generally plays nicer there).

Inside your ComfyUI venv:

C:\Users\phill\Documents\ComfyUI\.venv\Scripts\activate

pip uninstall -y torch torchvision torchaudio xformers

pip install --index-url https://download.pytorch.org/whl/cu118 \
  torch==2.1.2 torchvision==0.16.2

pip install xformers==0.0.22.post7


Driver: ensure an NVIDIA driver that supports CUDA 11.8 (522.xx or newer typically OK while still working on Maxwell).

If xformers gives you trouble on Maxwell, just skip it; use PyTorch attention (as you are).

4) Make Comfy less aggressive with memory

Set a couple env vars (permanent user env vars or a small .bat you use to launch ComfyUI):

set PYTORCH_CUDA_ALLOC_CONF=max_split_size_mb:64,expandable_segments:True,garbage_collection_threshold:0.9
set CUDA_LAUNCH_BLOCKING=0


Only set CUDA_LAUNCH_BLOCKING=1 when you want accurate stacks for debugging; it will slow everything down.

In ComfyUI, keep “pytorch attention” if xformers isn’t stable; VAE decode can be fp16 if available (yours logged fp32—fine, but fp16 saves a bit).

5) WAN-specific notes (your log hints)

Log shows “model weight dtype torch.bfloat16, manual cast: torch.float32”. Maxwell lacks bf16/tensor cores, so PyTorch promotes to fp32—much slower → longer kernels → triggers TDR. That’s why (1) and (2) are so important.

Your 121 frames @ 1280×704 is huge for 12 GB on Maxwell, even if VRAM seems free. The long per-step matmuls are what hit TDR.

If you must upscale: generate small + short, then do a video upscaler pass (e.g., frame-wise) after you have a working pipeline.

6) If it still trips:

Try DPM++ 2M Karras instead of UniPC.

Drop steps to 6–8, length 16–20, width/height one notch down (576×320).

Ensure desktop isn’t rendering heavy stuff while generating (same GPU runs display + kernels).

Update NVIDIA driver to a stable version for Maxwell (not the newest “bleeding edge” if it dropped Maxwell optimizations).

7) Debug mode (only if needed)

If you want a precise line for the hang:

set CUDA_LAUNCH_BLOCKING=1


Run once with tiny settings (e.g., 512×288, 8 steps, 12–16 frames) to capture a stack that points to the slow op, then turn it off again.

TL;DR start here

Change your node params to 672×384, 24 frames, steps 8, cfg 3.5, sampler euler, scheduler karras.

Import the TDR registry tweak and reboot.

Consider PyTorch 2.1.2 + cu118 (and maybe skip xformers on Maxwell).

Do those three and this particular “launch timed out” error should disappear; then you can scale back up bit-by-bit (width, frames, steps) until you find your GPU’s stable ceiling.

Hope this helps
